tests/test_levelset.py

```diff
@@ -0,0 +1,93 @@
+import unittest
+import firedrake as fd
+import fireshape as fs
+import fireshape.zoo as fsz
+import ROL
+
+class LevelsetTest(unittest.TestCase):
+
+    def run_levelset_optimization(self, Q, write_output=False):
+        """ Test template for fsz.LevelsetFunctional."""
+
+        #tool for developing new tests, allows storing shape iterates
+        if write_output:
+            out = fd.File("T.pvd")
+
+            def cb(*args):
+                out.write(Q.T)
+
+            cb()
+        else:
+            cb = None
+
+        # levelset test case
+        (x, y) = fd.SpatialCoordinate(Q.mesh_m)
+        f = (pow(x-0.5, 2))+pow(y-0.5, 2) - 2.
+        J = fsz.LevelsetFunctional(f, Q, cb=cb)
+
+        q = fs.ControlVector(Q)
+
+        # ROL parameters
+        params_dict = {
+            'General': {
+                'Secant': {'Type': 'Limited-Memory BFGS',
+                           'Maximum Storage': 25}},
+            'Step': {
+                'Type': 'Line Search',
+                'Line Search': {'Descent Method': {
+                    'Type': 'Quasi-Newton Step'}}},
+            'Status Test': {
+                'Gradient Tolerance': 1e-7,
```

Comment:
this convergence conditions are too tight for a unit test, suggest to set Gradient Tolerance to 1e-4 and Relative Gradien Tolerance to 1e-3